classes/robot/crawler.php

```diff
@@ -347,7 +347,7 @@ public function mark_for_crawl($baseurl, $url, $courseid = null, $priority = TOO
         }
 
         // Find the current node in the queue.
-        $node = url::get_record(['url' => $url]);
+        $node = url::get_record(['urlhash' => md5($url)]);
```

Comment:
can you please refactor a shared hash util so that we can change the url -> hash function in a single place

Reply:
Done in the latest commit